--- hostname.rb.orig	2013-07-26 18:38:17.967439708 +0000
+++ hostname.rb	2013-07-30 21:01:34.662409169 +0000
@@ -1,8 +1,10 @@
 #
 # Author:: Benjamin Black (<nostromo@gmail.com>)
 # Author:: Daniel DeLeo <dan@kallistec.com>
+# Author:: Jacques Marneweck <jacques@powertrip.co.za>
 # Copyright:: Copyright (c) 2008 Opscode, Inc.
 # Copyright:: Copyright (c) 2009 Daniel DeLeo
+# Copyright:: Copyright (c) 2013 Jacques Marneweck
 # License:: Apache License, Version 2.0
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
@@ -29,6 +31,12 @@
   # we recieved an fqdn
   fqdn fqdn_lookup
 else
-  # default to assembling one
-  fqdn(from("hostname") + "." + from("domainname"))
+  # nis?
+  if File.exists?("/etc/domainname")
+    # default to assembling one
+    fqdn(from("hostname") + "." + from("domainname"))
+  else
+    # Use the hostname
+    fqdn(from("hostname"))
+  end
 end
